recipe/meta.yaml

```diff
@@ -1,93 +1,109 @@
-    - $RPATH/libLLVM-14-rust-1.64.0-stable.so
-    # Since 1.32.0 macOS also needs:
-    - /System/Library/Frameworks/Python.framework/Versions/2.7/Python
-    - /System/Library/PrivateFrameworks/DebugSymbols.framework/Versions/A/DebugSymbols
-    - /usr/lib/libcompression.dylib
-    - /usr/lib/libedit.3.dylib
-    - /usr/lib/libform.5.4.dylib
-    - /usr/lib/libncurses.5.4.dylib
-    - /usr/lib/libpanel.5.4.dylib
-    - /usr/lib/libxml2.2.dylib
-    # Fixed via install_name_tool now (just declaring it here meant the thing using -
-    # lib/rustlib/x86_64-apple-darwin/lib/python2.7/site-packages/lldb/lldb-argdumper - was
-    # still unable to find it at runtime).
-    # - $RPATH/libLLVM.dylib
-
-    - '**/ld-linux-x86-64.so.2'
-    - '**/libc.so.6'
-    - '**/libdl.so.2'
-    - '**/ld64.so.*'
-    - '**/libgcc_s.so.1'
-    - '**/libm.so.6'
-    - '**/libpthread.so.0'
-    - '**/librt.so.1'
-    - '**/libgcc_s.so.1'
-    - '**/libz.so.1'
-    # Since 1.32.0 linux also needs:
-    - '**/libstdc++.so.6'
 
 outputs:
   - name: rust
     script: install-msvc.bat  # [win]
     script: install-unix.sh  # [not win]
+    build:
+      run_exports: # [osx and x86_64]
+        strong_constrains: # [osx and x86_64]
+          - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.12") }}  # [osx and x86_64]
+      # the distributed binaries are already relocatable
+      binary_relocation: False
+      runpath_whitelist:
+        - $ORIGIN/../lib
+      missing_dso_whitelist:
+        - /usr/lib/libresolv.9.dylib
+        - /usr/lib/libc++.1.dylib
+        - /usr/lib/libc++abi.dylib
+        - /usr/lib/libiconv.2.dylib
+        - /usr/lib/libcurl.4.dylib
+        - /usr/lib/libxar.1.dylib
+        - $RPATH/libLLVM-14-rust-1.64.0-stable.so
+        # Since 1.32.0 macOS also needs:
+        - /System/Library/Frameworks/Python.framework/Versions/2.7/Python
+        - /System/Library/PrivateFrameworks/DebugSymbols.framework/Versions/A/DebugSymbols
+        - /usr/lib/libcompression.dylib
+        - /usr/lib/libedit.3.dylib
+        - /usr/lib/libform.5.4.dylib
+        - /usr/lib/libncurses.5.4.dylib
+        - /usr/lib/libpanel.5.4.dylib
+        - /usr/lib/libxml2.2.dylib
+        # Fixed via install_name_tool now (just declaring it here meant the thing using -
+        # lib/rustlib/x86_64-apple-darwin/lib/python2.7/site-packages/lldb/lldb-argdumper - was
+        # still unable to find it at runtime).
+        # - $RPATH/libLLVM.dylib
+
+        - '**/ld-linux-x86-64.so.2'
+        - '**/libc.so.6'
+        - '**/libdl.so.2'
+        - '**/ld64.so.*'
+        - '**/libgcc_s.so.1'
+        - '**/libm.so.6'
+        - '**/libpthread.so.0'
+        - '**/librt.so.1'
+        - '**/libgcc_s.so.1'
+        - '**/libz.so.1'
+        # Since 1.32.0 linux also needs:
+        - '**/libstdc++.so.6'
     requirements:
       build:
         - {{ compiler('c') }}    # [osx]
         - posix  # [win]
     test:
+      requires:
+        - {{ compiler('c') }}
```

Comment:
what is a c compiler necessary for?

Reply:
To run forge_test.sh, a backend compiler is required.
Usually this is brought in by the activation feedstock, but here we are one step early, so we need to include it in test/requires.